### ISSUE TITLE
[Discussion] Remove dependency on ngrx store (in particular on ActionSubject)

### ISSUE THREAD

**Issue by mucaho** [MrWolfZ/ngrx-forms#165]
### Intro
First of all thanks for coming up with this amazing library, it's just the perfect tool for handling the dependencies between controls  - how updates of one control impacts values, validation, etc. on other controls.

### Description

Every directive has ngrx's `ActionSubject` injected into its constructor which is used to dispatch actions to the store (e.g. [the control's directive](https://github.com/MrWolfZ/ngrx-forms/blob/master/src/control/directive.ts)).
I would like to propose adding another directive that makes the output of actions explicit:

In your component you could listen for those actions and update the local component's state yourself,
dispatching any additional data loading or validation asynchronous requests:
```ts
// my-component.ts
@Component({
  selector: 'my-component',
  template: `
    <form novalidate
        [ngrxFormState]="(formState$ | async)"
        (ngrxFormState)="handleAction($event)">

    <input type="text"
        [ngrxFormControlState]="(formState$ | async).controls.countryCode"
        (ngrxFormControlState)="handleAction($event)">

    <input type="text"
        [ngrxFormControlState]="(formState$ | async).controls.country"
        (ngrxFormControlState)="handleAction($event)">

    </form>
  `,
})
export class MyComponent {
    const formState: FormGroupState<MyFormValue> = undefined;

    constructor(private actionSubject: ActionSubject) {
    }

    ngOnInit() {
        this.actionSubject.subscribe(action => {
            this.formState = myComponentReducer(this.formState, action);
        });
    }

    handleAction(action: Action) {
        this.actionSubject.next(action);
        if (action.payload.controlId === 'countryCode') {
            // this triggers loading in effect
            this.actionSubject.next(new RequestCountry({ countryCode: action.payload.value }));
        }
    }
}
```
```ts
// reducer.ts - although I would even inline this into component
const initialState: ComponentState = {
  loadCountryForCountryCode: null,
  myForm: initialFormState
};

export function myComponentReducer(state = initialState, action: Action): ComponentState {
    const myForm = formGroupReducer(state.myForm, action);
    if (myForm !== state.myForm) {
        state = { ...state, myForm };
    }

    switch (action.type) {
        // response from effect
        case RespondCountry: {
            state = {
                ...state,
                myForm = formStateReducer(state.myForm, new SetValueAction('country', action.payload.value))
            }
            return state;
        }
        // ...
        default: {
            return state;
        }
    }
}
```

### Motivation

I think is the missing link to the ngrx-ecosystem:
* You have **local**, component state which perishes after component is destroyed and does not taint global state.
* You have a way of dealing with form state that can become quite complex and inter-dependent using **easy-to-reason** and **easy-to-test** pure functions (read: reducers).
* You often use forms in modal dialog popups that cease to exist after they are submitted to server.
Global state is updated after the request returns from server with the entity you just created / updated, which you now see in your list, table, etc.

### Implementation
As easy as wrapping the current directives into new directives,
while passing an `@EventEmitter` to the constructor of the original directive.

### Conclusion
What do you think?

**Comment by MrWolfZ** [comment]
Yeah, makes sense. In fact, I started doing something similar once by extracting all the form state logic into a [separate library](https://github.com/MrWolfZ/pure-forms) that doesn't depend on Angular. The idea was to then write specific adapters for each framework, but also to allow using it directly for local component state as you did. I even used it like this in one of my projects and it worked quite well (the usage was basically exactly what you are suggesting with having a local reducer in the component).

I haven't updated that fork in some time, so there's probably some drift compared to **ngrx-forms** itself, but that should be simple enough to fix. However, I am not 100% certain how I could add this to **ngrx-forms** itself without breaking the existing behavior (which all consumers of course depend on).

Thinking about it, this could be a property on the form state itself set when you create it. And then in the directive based on this property the actions are either emitted on an `EventEmitter` or on the global `ActionsSubject`.

**Comment by mucaho** [comment]
Great, PR coming right up with `[createLocalFormState]` directive input

**Comment by MrWolfZ** [comment]
Resolved through #166
